qiskit/synthesis/multi_controlled/mcx_synthesis.py

```diff
@@ -300,6 +301,308 @@ def synth_mcx_noaux_v24(num_ctrl_qubits: int) -> QuantumCircuit:
     return qc
 
 
+def linear_depth_ladder_ops(qreg: List[int]) -> Tuple[QuantumCircuit, List[int]]:
```

Comment:
It would be nice if new code uses the standard library Python typehints, you can enable this on all our supported Python versions by adding
```
from __future__ import annotations
```
and then we don't need the `typing` module for hints.
```suggestion
def linear_depth_ladder_ops(qreg: list[int]) -> tuple[QuantumCircuit, list[int]]:
```